Tests/UserDataCacheSpec.swift

```diff
@@ -160,6 +160,47 @@ class UserDataCacheSpec: QuickSpec {
+                    let userA = [UserIdentifier(type: .idTrackingIdentifier, identifier: "idTrackingId1")]
+                    let userB = [UserIdentifier(type: .idTrackingIdentifier, identifier: "idTrackingId2")]
+                    userCache.cacheCampaignData(campaignsDataA, userIdentifiers: userA)
+                    userCache.cacheCampaignData(campaignsDataB, userIdentifiers: userB)
+
+                    expect(userCache.getUserData(identifiers: userA)).toNot(equal(userCache.getUserData(identifiers: userB)))
+                }
+
+                it("will use separate containers for the same userId with different idTrackingId") {
+                    let userA = [UserIdentifier(type: .userId, identifier: "userId1"),
+                                 UserIdentifier(type: .idTrackingIdentifier, identifier: "idTrackingId1")]
+                    let userB = [UserIdentifier(type: .userId, identifier: "userId1"),
+                                 UserIdentifier(type: .idTrackingIdentifier, identifier: "idTrackingId2")]
+                    userCache.cacheCampaignData(campaignsDataA, userIdentifiers: userA)
+                    userCache.cacheCampaignData(campaignsDataB, userIdentifiers: userB)
+
+                    expect(userCache.getUserData(identifiers: userA)).toNot(equal(userCache.getUserData(identifiers: userB)))
+                }
+
+                it("will use separate containers for the same rakutenId with different idTrackingId") {
+                    let userA = [UserIdentifier(type: .rakutenId, identifier: "rakutenId1"),
+                                 UserIdentifier(type: .idTrackingIdentifier, identifier: "idTrackingId1")]
+                    let userB = [UserIdentifier(type: .rakutenId, identifier: "rakutenId1"),
+                                 UserIdentifier(type: .idTrackingIdentifier, identifier: "idTrackingId2")]
+                    userCache.cacheCampaignData(campaignsDataA, userIdentifiers: userA)
+                    userCache.cacheCampaignData(campaignsDataB, userIdentifiers: userB)
+
+                    expect(userCache.getUserData(identifiers: userA)).toNot(equal(userCache.getUserData(identifiers: userB)))
+                }
+
+                it("will use separate containers for the same idTrackingId if also other id is present in one of them") {
+                    let userA = [UserIdentifier(type: .userId, identifier: "userId1"),
+                                 UserIdentifier(type: .idTrackingIdentifier, identifier: "idTrackingId1")]
+                    let userB = [UserIdentifier(type: .userId, identifier: "userId1")]
+                    userCache.cacheCampaignData(campaignsDataA, userIdentifiers: userA)
+                    userCache.cacheCampaignData(campaignsDataB, userIdentifiers: userB)
+
+                    expect(userCache.getUserData(identifiers: userA)).toNot(equal(userCache.getUserData(identifiers: userB)))
+                }
             }
```

Comment:
Is the `(!(preference.idTrackingIdentifier != nil && preference.accessToken != nil)` condition unit testable?

Reply:
The assertion is disabled in test environment. I'll try to make this testable